### PR TITLE
Display message at registration.

### LINKED ISSUE #1037: Check if banned members competed when they could not
We do have a list of banned people (and how long the ban lasts, of course), but we don't have a way to check this on the posting results process.

### PR DESCRIPTION
**Pull request by pedrosino**
Part of #1037 
(the diff looks ugly :P)

I initially thought of redirecting to the competitions index, but people might miss the alert at the top, so I decided to stay at the /register page.

Screenshot:
banned

### CODE REVIEW

File: WcaOnRails/db/structure.sql [hunk 1]
```diff
@@ -782,7 +782,8 @@ CREATE TABLE `registrations` (
   `accepted_by` int(11) DEFAULT NULL,
   `deleted_at` datetime DEFAULT NULL,
   `deleted_by` int(11) DEFAULT NULL,
-  PRIMARY KEY (`id`)
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `index_registrations_on_competition_id_and_user_id` (`competition_id`,`user_id`)
```
Comment (jfly) [inline WcaOnRails/db/structure.sql]: Hmm... why did this file change without a corresponding migration?

File: WcaOnRails/app/views/registrations/register.html.erb [hunk 2]
```diff
@@ -1,107 +1,111 @@
...
-  <% opens_in_days = distance_of_time_in_words_to_now(@competition.registration_open) %>
-  <% closes_in_days = distance_of_time_in_words_to_now(@competition.registration_close) %>
-  <% unless @competition.registration_opened? %>
-    <% if @competition.registration_not_yet_opened? %>
-      <%= alert :info, t('registrations.will_open_html', days: opens_in_days, time: wca_local_time(@competition.registration_open)) %>
-    <% end %>
-    <% if @competition.registration_past? %>
-      <%= alert :danger, t('registrations.closed_html', days: closes_in_days, time: wca_local_time(@competition.registration_close)) %>
-    <% end %>
+  <% if current_user.team_member?('banned') %>
```
Comment (jfly) [inline WcaOnRails/app/views/registrations/register.html.erb]: Can we pull this logic out into the `user` class? I'd rather see nice friendly methods like `banned?` and `banned_until` (`banned_until` might be sufficient... if `banned_until` is `nil`, then you're not banned).

Also, this diff is terrifyingly large, it's really hard to tell what happened here. Is there any reason we can't make use of our existing `cannot_register_for_competition_reasons` infrastructure instead, and leave this `erb` file alone?

**Comment by pedrosino** [comment]
Sorry it took so long to reply.

I feel writing php code to check this is going to be a big pain, and probably not worth the effort.
I agree with the concern about WDC being able to edit this "team" (and not the others (?))

Maybe this should wait a little longer in the end.

**Comment by jfly** [comment]
> Maybe this should wait a little longer in the end.

I agree. There's no real rush to implement this, so let's wait until the relevant PHP code has been ported to Rails, and we can do this right. I'm going to close this PR.

**Pull request closed without merging**